### PR TITLE
[feature/OPT-761] to dev

### PR DESCRIPTION
**Pull request by rmenciac**

### CODE REVIEW

File: otm/resources/schemas/otm_schema.json [hunk 1]
```diff
@@ -7,40 +7,40 @@
     "required": ["project", "otmVersion"],
     "properties": {
         "project": {
-            "type": "object",
+            "type": ["object", "null"],
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:10]: "project" is a required field, it shouldn't be null.

File: otm/resources/schemas/otm_schema.json [hunk 2]
```diff
@@ -49,18 +49,18 @@
             "properties": {
                 "name": {"type": "string"},
                 "id": {"type": "string"},
-                "description": {"type": "string"},
+                "description": {"type": ["string", "null"]},
                 "risk": {
-                    "type": "object",
+                    "type": ["object", "null"],
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:54]: "risk" is a required field so it shouldn't be null, instead "assets" is optional si a null should be allowed (line 47)

File: otm/resources/schemas/otm_schema.json [hunk 3]
```diff
@@ -49,18 +49,18 @@
...
                         "availability": {"type": "number"},
-                        "comment": {"type": "string"}
+                        "comment": {"type": ["string", "null"]}
                     }
                 },
-                "attributes": {"type": "object"}
+                "attributes": {"type": ["object", "null"]}
             }
         },
         "trustZones": {
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:66]: "trustZones" is an optional field so a null value should be allowed. Its "parent" field should also be optional (is this intended?)

File: otm/resources/schemas/otm_schema.json [hunk 4]
```diff
@@ -85,7 +85,7 @@
                         "type": "array",
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:85]: "representations" is an optional field for trustzones

File: otm/resources/schemas/otm_schema.json [hunk 5]
```diff
@@ -85,7 +85,7 @@
...
                         "items": {"$ref": "#/definitions/representationElement"}
                     },
-                    "attributes": {"type": "object"}
+                    "attributes": {"type": ["object", "null"]}
                 }
             }
         },
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:91]: "components" is an optional field so a null value should be allowed.

File: otm/resources/schemas/otm_schema.json [hunk 6]
```diff
@@ -98,40 +98,40 @@
...
                         "items": {
-                            "type": "string"
+                            "type": ["string", "null"]
                         }
                     },
-                    "attributes": {"type": "object"}
+                    "attributes": {"type": ["object", "null"]}
                 }
             }
         },
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:137]: "dataflows" is an optional field so a null value should be allowed.

File: otm/resources/schemas/otm_schema.json [hunk 7]
```diff
@@ -143,22 +143,25 @@
                 "properties": {
                     "id": {"type": "string"},
                     "name": {"type": "string"},
-                    "description": {"type": "string"},
+                    "description": {"type": ["string", "null"]},
                     "bidirectional": {"type": "boolean"},
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:147]: "bidirectional" is an optional field.

File: otm/resources/schemas/otm_schema.json [hunk 8]
```diff
@@ -143,22 +143,25 @@
...
                         "items": {
-                            "type": "string"
+                            "type": ["string", "null"]
                         }
                     },
-                    "attributes": {"type": "object"}
+                    "attributes": {"type": ["object", "null"]}
                 }
             }
         },
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:167]: "threats" is an optional field so a null value should be allowed.

File: otm/resources/schemas/otm_schema.json [hunk 9]
```diff
@@ -190,12 +193,12 @@
                         }
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:193]: "likelihoodComment" and "impactComment" are both optional fields.

File: otm/resources/schemas/otm_schema.json [hunk 10]
```diff
@@ -190,12 +193,12 @@
...
                         "items": {
-                            "type": "string"
+                            "type": ["string", "null"]
                         }
                     },
-                    "attributes": {"type": "object"}
+                    "attributes": {"type": ["object", "null"]}
                 }
             }
         },
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:204]: "mitigations" is an optional field so a null value should be allowed.

File: otm/resources/schemas/otm_schema.json [hunk 11]
```diff
@@ -207,9 +210,9 @@
...
                     "id": {"type": "string"},
                     "name": {"type": "string"},
-                    "description": {"type": "string"},
+                    "description": {"type": ["string", "null"]},
                     "riskReduction": {"type": "number"},
-                    "attributes": {"type": "object"}
+                    "attributes": {"type": ["object", "null"]}
                 }
             }
         }
```
Comment (galbiach-irius) [inline otm/resources/schemas/otm_schema.json:218]: "size" and "position" are optional for some representation types, so in those cases, they should be allowed to be null

**Review by galbiach-irius — DISMISSED** [review]

**Review by PacoCid — APPROVED** [review]

**Review by PacoCid — APPROVED** [review]

**Pull request merged**